pkg/kv/kvserver/replica_app_batch.go

```diff
@@ -223,6 +225,8 @@ func (b *replicaAppBatch) runPreApplyTriggersBeforeStagingWriteBatch(
 // runPreApplyTriggersAfterStagingWriteBatch runs any triggers that must fire
 // before a command is applied to the state machine but after the command is
 // staged in the replicaAppBatch's write batch.
+//
+// All errors from this method are fatal for this Replica.
```

Comment:
Whether it is true or not for any error depends on the scope. It's definitely not true in general throughout CRDB, as I'm sure you weren't implying. It is true for most below-raft code. So yes, this probably doesn't bear mentioning here and if we did it we'd have to do it consistently anywhere. The reason I still did it, albeit half-heartedly, is that it is important to keep it in mind. If we wrote most of this code with recoverability from errors in mind, it'd become orders of magnitude more complex (and probably still wouldn't be correct), because of the many modifications of in-memory state that are necessary e.g. for a split (but even easier cases would be difficult to unroll).

I'm going to remove these mentions and find one central place where this can be mentioned (maybe `Stage` on the interface), let's briefly touch upon this in sync communication to see if there's anything else we can do.

Reply:
Removed the mention, and `apply.Batch` already spells it out.